include/parser_interface.hpp

```diff
@@ -39,6 +39,26 @@ class ParserInterface
         return types::DbusVariantType();
     }
 
+    /**
+     * @brief API to write keyword's value on hardware.
+     *
+     * This virtual method is created to achieve runtime polymorphism for
+     * hardware writes on VPD of different types. This virtual method can be
+     * redefined at derived classess to implement write according to the type of
+     * VPD.
+     *
+     * @param[in] i_paramsToWriteData - Data required to perform write.
+     *
+     * @return On success returns number of bytes written on hardware, On
+     * failure returns -1.
```

Comment:
will this function throw ?

Reply:
this is just the virtual method. the derived class method which overrides it will throw.